entrypoint.rb

```diff
@@ -17,31 +19,44 @@ def query_check_status(ref, check_name, token)
     http.request(request)
   }
   parsed = JSON.parse(response.body)
-  return [nil, nil] if parsed["total_count"].zero?
 
-  [
-    parsed["check_runs"][0]["status"],
-    parsed["check_runs"][0]["conclusion"]
-  ]
+  parsed["check_runs"]
+end
+
+def reduce_checks_complete(checks)
```

Comment:
Like the `reduce` approach personally, but perhaps using `.any?` (https://rubyapi.org/2.7/o/enumerable#method-i-any-3F) or `.all?` will be more idiomatic and readable. Wanna give it another try?

Reply:
Great suggestion, thanks! I will amend now

Reply:
Done. The check for all checks being complete was so much more readable, and as it's not used more than once, I decided to inline it 🙂